src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -145,6 +223,10 @@ class OwlViTObjectDetectionOutput(ModelOutput):
         vision_model_last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_patches + 1, hidden_size)`)):
             Last hidden states extracted from the [`OwlViTVisionModel`]. OWL-ViT represents images as a set of image
             patches where the total number of patches is (image_size / patch_size)**2.
+        text_model_output (Tuple[`BaseModelOutputWithPooling`]):
```

Comment:
Can we add a deprecation warning to remove `text_model_last_hidden_state` and `vision_model_last_hidden_state` in the future?

Also, `image_embeds` and `class_embeds` should always return the projected last hidden states (similar to CLIP).

Reply:
Yes, I'm adding a deprecation warning to `OwlViTForObjectDetection.forward()`.`class_embeds` is specific to OwlViT and I think it makes more sense to return OwlViT  `image_embeds` instead of the unmodified CLIP embeddings